yarn-project/key-store/src/test_key_store.ts

```diff
@@ -277,49 +278,79 @@ export class TestKeyStore implements KeyStore {
   }
 
   /**
-   * Retrieves the master nullifier secret key (nsk_m) corresponding to the specified master nullifier public key
-   * (Npk_m).
+   * Retrieves the sk_m for the pk_m and a generator index of the key type.
    * @throws If the provided public key is not associated with any of the registered accounts.
-   * @param masterNullifierPublicKey - The master nullifier public key to get secret key for.
-   * @returns A Promise that resolves to the master nullifier secret key.
-   * @dev Used when feeding the master nullifier secret key to the kernel circuit for nullifier keys verification.
+   * @param masterPublicKey - The master public key to get secret key for.
+   * @returns A Promise that resolves to sk_m.
+   * @dev Used when feeding the sk_m to the kernel circuit for keys verification.
    */
-  public getMasterNullifierSecretKeyForPublicKey(masterNullifierPublicKey: PublicKey): Promise<GrumpkinPrivateKey> {
+  public getMasterSecretKeyAndAppKeyGenerator(masterPublicKey: PublicKey): Promise<[GrumpkinPrivateKey, KeyGenerator]> {
```

Comment:
I know this is quite ugly and stupid but I feel like it makes sense to come up with a more intelligent approach once most of the key stuff is in place since here the uglyness is quite isolated and it's all easy to test.

Reply:
Just did a large refactor of the key store so this comment is not relevant anymore.